pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -332,6 +356,11 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 			Fn:           flow.TaskFn(botanist.Shoot.Components.DNS.InternalEntry.Destroy).DoIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(hibernateControlPlane),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying Kubernetes API server service SNI settings in the Seed cluster",
+			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.KubeAPIServerSNI.Destroy).DoIf(o.Shoot.HibernationEnabled),
```

Comment:
Why it is not destroyed when SNI is disabled?

Reply:
It's was originally removed so the `LoadBalancer` is not used. With this change it's better to keep it as it's of type `ClusterIP`.

Reply:
My point is that if I enable and disable the SNI feature, I will have these orphaned resources.
Yes, they do cost nothing, but as long as the feature is alpha, this resource can be replaced by anything else.

Reply:
I've removed this.

Reply:
But I do not asked you to remove it.
The question was more to the `DoIf(o.Shoot.HibernationEnabled)` ?

Reply:
ping

Reply:
this changed is now in https://github.com/gardener/gardener/blob/3f0c10df007972e96117b2e2bd6c30e8d0e8bef8/pkg/operation/botanist/controlplane.go#L327-L329